### PR TITLE
#162600229 Build view all articles

### PR DESCRIPTION
**Pull request by tersoo-atsen**
#### What does this PR do?

Have a view where user can see all articles

#### Description of Task to be completed?

- add tests
- add AllArticlePage component
- extract PopularPosts Component

#### How should this be manually tested?

- checkout to this branch
- run `npm install` to install necessary dependencies
- run `npm run dev`
- navigate to `localhost:8080/articles/categories/:category`

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#162600229](https://www.pivotaltracker.com/story/show/162600229)

#### Screenshots (if appropriate)

screen shot 2018-12-12 at 7 42 31 pm
screen shot 2018-12-12 at 7 42 22 pm
screen shot 2018-12-12 at 7 52 50 pm
screen shot 2018-12-12 at 7 52 15 pm

#### Questions:

N/A

### CODE REVIEW

File: public/styles/HomePage.scss [hunk 1]
```diff
@@ -20,129 +20,26 @@
...
-            border-top: 1px solid $color-secondary;
-            color: $text-grey;
-            font-family: 'Zilla Slab';
-            font-size: 2em;
-            margin: 0 1em 0.4em 0;
-            padding: 0.5em 0;
-            text-align: center;
-            text-transform: capitalize;
-          }
+          overflow: hidden;
```
Comment (darthrighteous) [inline public/styles/HomePage.scss]: Overflow here should be visible. This is making the cards get cut on the hover effect

File: public/styles/allArticles.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
+.main {
+
+  #primary {
+
+    .category-cont {
+      overflow: hidden;
```
Comment (darthrighteous) [inline public/styles/allArticles.scss]: Overflow here should be visible as well. This is making the cards get cut on the hover effect

**Review by darthrighteous — CHANGES_REQUESTED** [review]
Please connect the Links in the nav bar to their respective categories page

**Review by sulenchy — APPROVED** [review]
LGTM

File: public/styles/HomePage.scss (hunk 1 above)
Comment (tersoo-atsen) [reply public/styles/HomePage.scss]: I have fixed this.

File: public/styles/allArticles.scss (hunk 2 above)
Comment (tersoo-atsen) [reply public/styles/allArticles.scss]: I have fixed this.

File: tests/src/components/Card.test.js [hunk 3]
```diff
@@ -20,3 +20,10 @@ test('CardList snapshot test', () => {
   expect(component).toMatchSnapshot();
   expect(component2).toMatchSnapshot();
 });
+
+test('CardList snapshot test', () => {
+  const component = shallow(<CardList article={articles} className="category-cont" title="testing" />);
+  const component2 = shallow(<CardList article={articles} className="testing" title="testing" />);
+  expect(component).toMatchSnapshot();
+  expect(component2).toMatchSnapshot();
```
Comment (augustineezinwa) [inline tests/src/components/Card.test.js]: Is there any reason why you named this variable: `component2`. Please use an appropriate variable name. Refer also to this: [https://github.com/airbnb/javascript/tree/es5-deprecated/es5#naming-conventions](url)

**Review by augustineezinwa — CHANGES_REQUESTED** [review]
Make the requested changes. The PR looks good.

**Review by darthrighteous — APPROVED** [review]

File: tests/src/components/Card.test.js (hunk 3 above)
Comment (tersoo-atsen) [reply tests/src/components/Card.test.js]: I have made these adjustments.

**Review by augustineezinwa — APPROVED** [review]

**Pull request merged**